### ISSUE TITLE
GRPO训练中 attention_mask 和 completion_mask 不对齐

### ISSUE THREAD

**Issue by RewindL** [modelscope/ms-swift#7066]
环境：ms_swift 3.11.0.dev0

操作：
我在 `grpo_trainer.py` 的 `_compute_loss_and_metrics#` 处打断点，并启动GRPO训练查看传入的数据结构。

断点位置：
[    def _compute_loss_and_metrics(self, model, inputs):](https://github.com/modelscope/ms-swift/blob/b3ab7b6c17c20fc8ce7b5fc7064ac2b1f0e678cc/swift/trainers/rlhf_trainer/grpo_trainer.py#L988)

问题：
我查看 inputs 传入的 input_ids, attention_mask 和 completion_mask 时出现了以下不对齐的问题。

```
(Pdb) completion_mask.shape
torch.Size([16, 2062])
(Pdb) inputs['input_ids'].shape
torch.Size([16, 2195])
(Pdb) inputs['logits_to_keep']
2062
```
这里 completion_mask 不含 prompt部分，所以会短一点。

但是我继续打印第一条数据的信息时：
```
(Pdb) inputs['input_ids'][0,-inputs['logits_to_keep']:][:100]
tensor([   323,   2182,    697,   1590,   4226,  ... , 17,  11035,   1291,      8, 1124], device='cuda:0')
(Pdb) inputs['attention_mask'][0,-inputs['logits_to_keep']:][:100]
tensor([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ... , 1, 1, 1, 1, 1, 1], device='cuda:0')
(Pdb) inputs['completion_mask'][0,-inputs['logits_to_keep']:][:100]
tensor([False, False, False, False, False, False, False, False, False, False,
        False, False, False, False,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True],
       device='cuda:0')
```
我发现 `input_ids` 和 `attention_mask` 的 prompt 和 completion 都是紧贴着且左对齐右padding的，但是`completion_mask` 前面却有一截左 padding。

请问这是正常的吗？为什么会出现这种情况呢？

**Comment by hjh0119** [comment]
正常的，logits_to_keep 减少获取logits时的运算成本，可以搜一下相关代码

loss 计算里的 tensor seq_len 都是 logits_to_keep
